### PR TITLE
Even if the screen orientation is locked, the device will render corr…

### PR DESCRIPTION
**Pull request by Diolor**
Even if the screen orientation is locked, the device will render correctly the image rotation.

### CODE REVIEW

**Comment by Diolor** [comment]
Based on #90

**Comment by jpribble** [comment]
I think this PR may have introduced a regression. I've seen an issue occur a couple times in the past few days where logcat continuously repeats the following logs very quickly:

> 01-25 14:40:59.920 31137-32195/ D/Fotoapparat: CameraDevice: setDisplayOrientation
> 01-25 14:40:59.920 31137-32195/ D/Fotoapparat: Image orientation is: io.fotoapparat.hardware.orientation.Orientation$Vertical$Portrait@f012458. 
>                                                                        Display orientation is: io.fotoapparat.hardware.orientation.Orientation$Vertical$Portrait@f012458. 
>                                                                        Preview orientation is: io.fotoapparat.hardware.orientation.Orientation$Vertical$Portrait@f012458.
> 01-25 14:40:59.962 31137-32195/ D/Fotoapparat: CameraDevice: setDisplayOrientation
> 01-25 14:40:59.962 31137-32195/ D/Fotoapparat: Image orientation is: io.fotoapparat.hardware.orientation.Orientation$Horizontal$ReverseLandscape@7dfc53b. 
>                                                                        Display orientation is: io.fotoapparat.hardware.orientation.Orientation$Vertical$Portrait@f012458. 
>                                                                        Preview orientation is: io.fotoapparat.hardware.orientation.Orientation$Vertical$Portrait@f012458.
> 01-25 14:40:59.978 31137-32195/ D/Fotoapparat: CameraDevice: setDisplayOrientation
> 01-25 14:40:59.978 31137-32195/ D/Fotoapparat: Image orientation is: io.fotoapparat.hardware.orientation.Orientation$Vertical$Portrait@f012458. 
>                                                                        Display orientation is: io.fotoapparat.hardware.orientation.Orientation$Vertical$Portrait@f012458. 
>                                                                        Preview orientation is: io.fotoapparat.hardware.orientation.Orientation$Vertical$Portrait@f012458.
> 01-25 14:41:00.035 31137-32195/ D/Fotoapparat: CameraDevice: setDisplayOrientation
> 01-25 14:41:00.035 31137-32195/ D/Fotoapparat: Image orientation is: io.fotoapparat.hardware.orientation.Orientation$Horizontal$ReverseLandscape@7dfc53b. 
>                                                                        Display orientation is: io.fotoapparat.hardware.orientation.Orientation$Vertical$Portrait@f012458. 
>                                                                        Preview orientation is: io.fotoapparat.hardware.orientation.Orientation$Vertical$Portrait@f012458.
> ...

and occasionally the camera preview images get distorted. The image appears stretched as if a landscape image is trying to be displayed as portrait. Do you know what could be causing this? I've seen it occur on a Samsung GS8 and GS6 with the device's orientation locked to portrait and the camera activity's orientation locked to landscape.

**Comment by jpribble** [comment]
@Diolor Would it be possible to add a configuration to enable/disable this _fix_? For my use case, if the camera activity is locked to landscape, I expect the photos to always be delivered with landscape orientation. If the user holds the device flat (parallel with the surface of the ground) the device's orientation is unreliable and will cause photos to be rotated unexpectedly.

**Comment by Diolor** [comment]
Regarding the logcat above, might be the orentation listener. Maybe we could improve this.

About the orientation when in locked (reverse) landscape, I actually had not tested it. Only the portrait mode. Might be related to #193

File: fotoapparat/src/main/java/io/fotoapparat/hardware/display/Display.kt [hunk 1]
```diff
@@ -12,14 +17,14 @@ internal open class Display(context: Context) {
...
-        Surface.ROTATION_90 -> 90
-        Surface.ROTATION_180 -> 180
-        Surface.ROTATION_270 -> 270
-        Surface.ROTATION_0 -> 0
-        else -> 0
+    open fun getOrientation(): Orientation = when (display.rotation) {
+        Surface.ROTATION_0 -> Portrait
+        Surface.ROTATION_90 -> Landscape
+        Surface.ROTATION_180 -> ReversePortrait
+        Surface.ROTATION_270 -> ReverseLandscape
```
Comment (jpribble) [inline fotoapparat/src/main/java/io/fotoapparat/hardware/display/Display.kt]: This looks backwards to me. When I tested this on Samsung GS8, 90 was ReverseLandscape and 270 was Landscape.

File: fotoapparat/src/main/java/io/fotoapparat/hardware/orientation/Rotation.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+typealias DeviceRotationDegrees = Int
+
+/**
+ * @return closest right angle to given value. That is: 0, 90, 180, 270.
+ */
+internal fun Int.toClosestRightAngle(): Int {
+    val roundUp = this % 90 > 45
+
+    val roundAppModifier = if (roundUp) 1 else 0
```
Comment (jpribble) [inline fotoapparat/src/main/java/io/fotoapparat/hardware/orientation/Rotation.kt]: Should this be "roundUp" instead of "roundApp"?

**Pull request merged**